collector/kernel/bpf_src/render_bpf.c

```diff
@@ -2321,21 +2321,63 @@ int on_cgroup_populate_dir(struct pt_regs *ctx, struct cgroup *cgrp, unsigned lo
 }
 #endif
 
-// existing
+// existing cgroups v2
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 6, 0)
+BEGIN_DECLARE_SAVED_ARGS(cgroup_control)
+struct cgroup *cgrp;
+END_DECLARE_SAVED_ARGS(cgroup_control)
+
+int on_cgroup_control(struct pt_regs *ctx, struct cgroup *cgrp)
+{
+  GET_PID_TGID;
+
+  BEGIN_SAVE_ARGS(cgroup_control)
+  SAVE_ARG(cgrp)
+  END_SAVE_ARGS(cgroup_control)
+
+  return 0;
+}
+
+int onret_cgroup_control(struct pt_regs *ctx)
+{
+  GET_PID_TGID;
+
+  GET_ARGS_MISSING_OK(cgroup_control, args)
+  if (args == NULL) {
+    return 0;
+  }
+
+  struct cgroup *cgrp = args->cgrp;
+
+  u16 subsys_mask = (u16)PT_REGS_RC(ctx);
+  if (!(subsys_mask & 1 << FLOW_CGROUP_SUBSYS))
+    return 0;
```

Comment:
Missing `DELETE_ARGS`? I think args can be deleted after getting the `args->cgrp`.

Reply:
The DELETE_ARGS is down at line 2361.  I'll move it up to make it more clear.

Reply:
What I meant was that the return on 2354 is missing a `DELETE_ARGS`.

Reply:
Gotcha, yes, that is a problem :^)